### PR TITLE
[Ci] clean up the old artifacts in the agent before downloading the new artifacts

### PR DESCRIPTION
**Pull request by xumia**
#### Why I did it
The vs build failed but some of the test passed.
See build: https://dev.azure.com/mssonic/build/_build/results?buildId=159753&view=logs&jobId=e2a1d004-6151-514d-dd3b-ba225f574bd9&j=e2a1d004-6151-514d-dd3b-ba225f574bd9&t=f0777b63-8377-5b65-c700-6568122e76b9

```
[ FAIL LOG START ] [ target/python-wheels/buster/sonic_yang_models-1.0-py3-none-any.whl ]
[ REASON ] :      target/python-wheels/buster/sonic_yang_models-1.0-py3-none-any.whl does not exist   NON-EXISTENT PREREQUISITES: target/debs/buster/libyang_1.0.73_amd64.deb-install target/debs/buster/libyang-cpp_1.0.73_amd64.deb-install target/debs/buster/python3-yang_1.0.73_amd64.deb-install 
[ FLAGS  FILE    ] : [] 
[ FLAGS  DEPENDS ] : [vs amd64 buster] 
[ FLAGS  DIFF    ] : [vs amd64 buster ] 
/sonic/src/sonic-yang-models /sonic
running pytest
Searching for ijson==2.6.1
Reading https://pypi.org/simple/ijson/

```

image

It is another issue, the vs build failed, but the job is succeeded, fixed in PR https://github.com/sonic-net/sonic-buildimage/pull/12369.

#### How I did it
Clean up the old artifacts.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -96,6 +96,11 @@ stages:
       submodules: recursive
       displayName: 'Checkout code'
 
+    - script: |
+        sudo rm -rf ../target
```
Comment (liushilongbuaa) [inline azure-pipelines.yml:100]: It's better to set '-e'

Comment (liushilongbuaa) [inline azure-pipelines.yml:100]: set -e
sudo rm -rf ../target

**Review by liushilongbuaa — CHANGES_REQUESTED** [review]

File: .azure-pipelines/run-test-template.yml [hunk 2]
```diff
@@ -26,6 +26,10 @@ steps:
   clean: true
   displayName: 'checkout sonic-mgmt repo'
 
+- script: |
+    sudo rm -rf ../target
```
Comment (xumia) [inline .azure-pipelines/run-test-template.yml:30]: sudo rm ../target

File: .azure-pipelines/run-test-template.yml [hunk 3]
```diff
@@ -26,6 +26,10 @@ steps:
...
+  displayName: "Cleanup"
```
Comment (xumia) [inline .azure-pipelines/run-test-template.yml:31]:
```suggestion
    sudo rm -rf ../target
  displayName: "Cleanup"
```

**Pull request merged**